binarySearch.js

```diff
@@ -0,0 +1,59 @@
+/*
+We're going to approach this problem using a binary search algorithm where we'll be able to search through
+a sorted list efficintly by repeatedly dividing a porition of the list in half, until we've narrow down the location
+of the target.
+
+We'll set two pointers, one at the start of the list and other at the end of the list.
+And from there, we'll do a comparison of values with the targert and depending
+on the outcome, we'll either move the left or right pointer until we're closer to the target.
+*/
+
+/*
+Question:
+Given an array of integers nums which is sorted in ascending order, and an integer target, write a function to search target in nums. 
+If target exists, then return its index. Otherwise, return -1.
+
+You must write an algorithm with O(log n) runtime complexity.
+*/
+const NOT_FOUND = -1;
+
+const getMidIx = (rightIdx, leftIdx) => {
+    const idxRange = rightIdx - leftIdx;
+    const midIdx = leftIdx + Math.floor(idxRange / 2);
```

Comment:
P: Why don't you just do `(leftIdx + rightIdx) // 2`?

Reply:
It's because if there was a larger range of numbers, it might be difficult to store a large value as a variable. Also, JavaScript does have a limit of storing values in memory.

Reply:
The key term here is "integer overflow". There's often a limit on the largest number that your language can store. It is guaranteed that your two inputs `rightIdx` and `leftIdx` are within the valid range for numbers. However, their sum might be outside of the range. So, it's possible that doing `(leftIdx + rightIdx)` will cause an integer overflow.

But, if you use the math that you did, then you never compute the sum of these numbers. So, there is no need to worry about integer overflow.

Why don't you try answering this question again?

Reply:
Ah, ok. Let me know what you think of my explanation:

Since JavaScript has a limit for the largest number that it can store, there's a possibility for an integer overflow. Even though the `rightIdx` and `leftIdx` are within range, adding them could cause them to be outside of that range. But by taking the difference of those two indices, dividing it by 2, flooring it and adding the `leftIdx`, we're able to avoid an integer overflow and still calculate the `midIdx`.

Reply:
Fantastic explanation!